wait/nop.go

```diff
@@ -0,0 +1,65 @@
+package wait
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/go-connections/nat"
+)
+
+type NopStrategy struct {
```

Comment:
Do we make this strategy into the public API? I'd suggest not doing it unless we see the value for the downstream users

Reply:
This would be available in public API for consumers to use with the existing strategies. I've seen various third-party and internal/std packages which expose a Nop version.

* https://pkg.go.dev/io#NopCloser
* https://pkg.go.dev/github.com/datadog/datadog-go/statsd#NoOpClient
* https://pkg.go.dev/go.uber.org/zap#NewNop

Reply:
Indeed. Let's keep the NopStrategy. Cloud you differentiate it from the already existing `noopStrategyTarget`, from the log_test.go file?

I'd keep just one.

Reply:
I went with `Nop` instead of `Noop` or `NoOp` since the standard library `io.NopCloser` used it, I tend to prefer what the standard library uses. I have no issues with changing it to `Noop` or `NoOp`

Reply:
Ok then, I'd like to stand with standard library too. Let's keep the prefix as `Nop`. Could you apply it everywhere?